packages/avocode-email-tagsinput/package.json

```diff
@@ -9,7 +9,7 @@
     "url": "git+https://github.com/avocode/avocode-email-tagsinput.git"
   },
   "scripts": {
-    "build": "npm run clean && npm run copy-public-flow-types && npm run flow:libdef && webpack --mode=production",
+    "build": "npm run clean && npm run copy-public-flow-types && npm run flow:libdef && babel  ./lib --out-dir ./dist --config-file ../avocode-email-tagsinput-config/lib/babel.config.js",
```

Comment:
You should not require the config file from parent directory. `avocode-email-tagsinput-config` is a dependency.

At least at the time when I designed the monorepo, I went with the approach of never touching other packages via relative paths / imports since they should be independent - including build.

Maybe better approach would be to use Babel programmatic API and use the config from the package itself via `import`.

Reply:
Ondra is right—you should not touch files outside the package.

The way to do this is to use the config package as a dependency.

Reply:
Aha, I see, wepback config is taken from other package.

I guess I'll make another package.

Reply:
But that's not idea either. Because afterward there are `utils` bundled with `avocode-tags-input`. There should not have been webpack used in `build` script at all. It does not make any sense. I gues that was used so that playground could have been released.

The scripts should be different for `build` to release the playground and for `prepublish` to just transpile the code to JS. @comatory

Reply:
So I will rather use special `babel.config.js` in every package and re-export it.

Reply:
Just be aware that `utils` in `avocode-emails-taginput` actually re-use utils from `better-react-tagsinput`.

And that util file is also dependent on `slate`. So that might also give you further complications for code splitting.

I'd either separate `slate` into some kind of chunk that can be re-used by other packages or create `util` package that lists `slate` as its peer  dependency so you don't end up bundling it. The latter is probably easier choice and should not be too difficult. Follow Lerna documentation on how to create new package, it has a script that generates some bootstrapping code.

Reply:
I don't remember all the details anymore so I'm not sure what happens if you don't transpile the code. Why don't you try it first by running `npm pack` with `babel` and see  if you can make it work in Avocode app.

Also having the playground working is pretty important for development IMHO as it'd be painful to do it without it so maybe just keep webpack around for that purpose.

I guess you could just make copies of babel.config.js everywhere but again, the config package is already preconfigured with everything.
I've seen monorepos where they keep config files somewhere in root and reuse them within packages but it depends on how "strict" you want to be with having everything separated cc @jankuca

Reply:
I am not, of course, removing playground :).

But apparantly `utils` use `slate` and it has got 40kB gzipped, so we won't save anything. So I am closing it. I will have to resolve it in the app.

Reply:
Yeah that's disappointing.  It was designed to come with at least Slate dependency.

However there could be easy solution. If you move `slate` and `react-slate` into peer dependencies and then include them inside the app, you should end up with light bundle and it should not be a lot of work to do this.

Reply:
Declaring these as peer dependencies should be "just" a matter of changing `package.json`. Lerna should then install them for development purposes.

Reply:
I don't think it will help because slate must be loaded + that `utils` have some more methods added.